### PR TITLE
Support/pin pr workflow deps

### PR DESCRIPTION
**Pull request by angusbayley**
### 📝 Description

Pins dependencies for all PR workflows. That means all workflows triggered `on: pull_request` or `on: push`.

### ❓ Context

Ticket: https://ledgerhq.atlassian.net/browse/LIVE-14091

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: .github/workflows/build-desktop-external-reusable.yml [hunk 1]
```diff
@@ -19,13 +19,18 @@ jobs:
   build-desktop-app:
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
-    name: "Build Ledger Live Desktop"
-    runs-on: ${{ matrix.os }}
```
Comment (angusbayley) [inline .github/workflows/build-desktop-external-reusable.yml:24]: this is a result of the workflows in `build-desktop-external-reusable.yml` and `build-desktop-reusable.yml` drifting out of sync, as they are partial copy-pasted duplicates of each other. Would love to remove as much duplication as possible (ideally delete the unused workflow)

Comment (KVNLS) [reply .github/workflows/build-desktop-external-reusable.yml:24]: I'm not sure. We don't want to use our internal runners for external contributions. So I think that ubuntu-latest is used for Github runner. Can you double check, please?

**Review by KVNLS — CHANGES_REQUESTED** [review]

File: .github/workflows/build-desktop-external-reusable.yml (hunk 1 above)
Comment (angusbayley) [reply .github/workflows/build-desktop-external-reusable.yml:24]: Good point - fixed

**Review by KVNLS — APPROVED** [review]
LGTM!

**Pull request merged**